src/Controls/src/Core/Border/Border.cs

```diff
@@ -309,16 +310,29 @@ protected override void OnPropertyChanged([CallerMemberName] string? propertyNam
 		{
 			base.OnPropertyChanged(propertyName);
 
-			if (propertyName == HeightProperty.PropertyName ||
-				propertyName == StrokeThicknessProperty.PropertyName ||
-				propertyName == StrokeShapeProperty.PropertyName ||
-				propertyName == WidthProperty.PropertyName)
+			if (propertyName == WidthProperty.PropertyName || propertyName == HeightProperty.PropertyName)
+			{
+				_sizeChanged = true;
+			}
+			else if (propertyName == StrokeThicknessProperty.PropertyName || propertyName == StrokeShapeProperty.PropertyName)
 			{
-				Handler?.UpdateValue(nameof(IBorderStroke.Shape));
 				UpdateStrokeShape();
+				Handler?.UpdateValue(nameof(IBorderStroke.Shape));
 			}
 			else if (propertyName == StrokeDashArrayProperty.PropertyName)
+			{
 				Handler?.UpdateValue(nameof(IBorderStroke.StrokeDashPattern));
+			}
+		}
+
+		protected override void OnSizeAllocated(double width, double height)
+		{
+			base.OnSizeAllocated(width, height);
+			if (_sizeChanged)
+			{
+				_sizeChanged = false;
+				Handler?.UpdateValue(nameof(IBorderStroke.Shape));
```

Comment:
is there a way this could move down into the handler?

OnSizeAllocated is basically called during the arrange pass when `Handler?.PlatformArrange(Frame);` is called.